jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/authentication/FormAuthenticator.java

```diff
@@ -360,17 +360,7 @@ private AuthenticationState dispatch(String path, Request request, Response resp
         {
             String newPath = URIUtil.addPaths(request.getContext().getContextPath(), path);
             HttpURI.Mutable newUri = HttpURI.build(request.getHttpURI()).pathQuery(newPath);
-            return new AuthenticationState.ServeAs(newUri)
-            {
-                @Override
-                public Request wrap(Request request)
-                {
-                    PathWrapper serveAs = Request.as(request, PathWrapper.class);
-                    if (serveAs != null)
-                        return serveAs.serveAs(request, path);
-                    return super.wrap(request);
-                }
-            };
+            return (AuthenticationState.ServeAs)req -> Request.serveAs(req, newUri);
```

Comment:
I kind of preferred the ServeAs carrying the HttpURI rather than just capturing it here.    It will debug better that way as well, as lambdas don't look good in debug.